src/rgw/driver/rados/rgw_cr_rados.cc

```diff
@@ -912,9 +914,55 @@ int RGWAsyncRemoveObj::_send_request(const DoutPrefixProvider *dpp)
     return 0;
   }
 
-  RGWAccessControlPolicy policy;
+  RGWObjTags obj_tags;
+  bufferlist bl_tag;
+  if (obj->get_attr(RGW_ATTR_TAGS, bl_tag)) {
+    auto bliter = bl_tag.cbegin();
+    try {
+      obj_tags.decode(bliter);
+    } catch (buffer::error &err) {
+      ldpp_dout(dpp, 0) << "ERROR: " << __func__ << ": caught buffer::error couldn't decode TagSet " << dendl;
+      return -EIO;
+    }
+  }
+
+  rgw_sync_pipe_params params;
+  if (!sync_pipe.info.handler.find_obj_params(obj->get_key(),
+                                              obj_tags.get_tags(),
+                                              &params)) {
+    return 0; // -ERR_PRECONDITION_FAILED
+  }
+
+  if (params.mode == rgw_sync_pipe_params::MODE_USER) {
+    std::optional<RGWUserPermHandler> user_perms;
+    RGWUserPermHandler::Bucket dest_bucket_perms;
+
+    if (params.user.empty()) {
+      ldpp_dout(dpp, 20) << "ERROR: " << __func__ << ": user level sync but user param not set" << dendl;
+      return 0; // -EPERM
```

Comment:
why do these error cases return 0?

Reply:
ops thanks, sorry my bad. I wasn't sure whether those checks for skipping the object https://github.com/ceph/ceph/blob/290894cc762024be7774bdda6a02815a1e08555e/src/rgw/driver/rados/rgw_data_sync.cc#L4565 was going to apply for delete replication as well. I checked now and it will so I modified those to reflect the actual error.